metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -329,12 +330,17 @@ def get_log_stream(job):
                         else:
                             pass
                     else:
+                        i = 0
                         yield line
+                return
             except Exception as ex:
+                exception = ex
                 if self.is_crashed:
                     break
-                sys.stderr.write(repr(ex))
-                time.sleep(2 ** i)
+                #sys.stderr.write(repr(ex) + '\n')
+                if i < self.NUM_RETRIES:
```

Comment:
Why is there no sleep on the last retry? Overall it was confusing for me to understand why you don't chose the constant to 9; instead of +1 on L:321.

Reply:
`NUM_ATTEMPTS = NUM_RETRIES + 1`. We would like the loop to execute `NUM_ATTEMPTS` times and we would immediately like to raise the exception after the last attempt - sleeping at that point is not helpful in any way.

Reply:
Let me know if you think this might help readability.
```
NUM_ATTEMPTS=9

for attempt in range(0, NUM_ATTEMPTS):
     try:
     // log processing (with reset on attempt)
     except:
     if attempt == NUM_ATTEMPTS - 1: # last attempt
        raise(..)
     else: # exponential back-off with jitter
       sleep(..)
```